shopinvader_address/models/res_partner.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import MissingError, UserError
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    def _shopinvader_billing_addresses_already_used(self):
+        """
+        Check if Address is used on posted account move
+        """
+        self.ensure_one()
+        move_id = (
+            self.env["account.move"]
+            .sudo()
+            .search(
+                [("commercial_partner_id", "=", self.id), ("state", "=", "posted")],
+                limit=1,
+            )
+        )
+        return len(move_id) > 0
+
+    # --- Billing ---
+    # Billing addresses is unique and corresponds to authenticated_partner
+
+    def _get_shopinvader_billing_addresses(self) -> "ResPartner":
+        self.ensure_one()
+        return self
+
+    def _update_shopinvader_billing_addresses(self, vals: dict) -> "ResPartner":
+        self.ensure_one()
+        # if billing addresses is already used, it is not possible to modify it
+        if self._shopinvader_billing_addresses_already_used():
+            raise UserError(
+                _(
+                    "Can not update billing addresses(%d) because it is already used on billings" % self.id
+                )
```

Comment:
```suggestion
                _(
                    "Can not update billing addresses (%(address_id)d) because it is already used on billings",
                    address_id=self.id
                )
```